src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const cleanedString = sourceString.replace(/;\s*;/g, ';');
```

Comment:
The regex used in 'replace' may not be necessary if the input string is well-formed. Ensure that input strings are always correctly formatted to avoid unnecessary operations.